java/lib/core/src/main/java/org/eclipse/tahu/model/MqttServerDefinition.java

```diff
@@ -27,16 +29,18 @@ public class MqttServerDefinition {
 	private final String password;
 	private final int keepAliveTimeout;
 	private final Topic ndeathTopic;
+	private final SSLSocketFactory socketFactory;
 
 	public MqttServerDefinition(MqttServerName mqttServerName, MqttClientId mqttClientId, MqttServerUrl mqttServerUrl,
-			String username, String password, int keepAliveTimeout, Topic ndeathTopic) {
+			String username, String password, int keepAliveTimeout, Topic ndeathTopic, SSLSocketFactory socketFactory) {
```

Comment:
Can you add a new constructor instead of modifying the existing one. I'd rather not break existing applications that use this.

Reply:
yes, of course! I forgot this might be used outside of this library :)